src/frontend/pages/enrich.vue

```diff
@@ -336,29 +391,41 @@ export default {
             responseType: 'json'
           })
           this.state = state?.data
-          // TODO use computed
-          if (Number.isInteger(this.state?.createdAt)) {
-            this.time = Math.round((Date.now() - this.state?.createdAt) / 1000)
-          } else {
-            this.time = 0
-          }
         } catch (err) {
           this.$store.dispatch('snacks/error', this.$t('enrich.errorState'))
           return this.errored()
         }
         await new Promise(resolve => setTimeout(resolve, 1000))
       } while (!state?.data?.done)
 
+      this.stopTimer()
       // done
       this.inProcess = false
       this.id = data.id
     },
 
+    startTimer () {
+      this.timer = setInterval(() => {
+        this.time += 1
```

Comment:
Incrementing seconds is not very reliable, as `setInterval()` and `setTimeout()` are not guaranted to be accurate, and the Javascript engine can be paused by the browser when switching to another tab. Using dates is much more reliable when it gets to creating a timer (ie. memorize start date and compute the difference with current date).